readthedocs/core/validators.py

```diff
@@ -44,4 +45,19 @@ def __call__(self, value):
                 raise exc
             super(DomainNameValidator, self).__call__(idnavalue)
 
+
 validate_domain_name = DomainNameValidator()
+
+
+def validate_spam(value):
+    """
+    Assume that anything with a 1 and 8 in it is spam.
+
+    We have had a ton of phone number spam,
+    and I think taking a large swath approach will be good for now.
+    """
+
+    if '1' in value and '8' in value:
+        raise ProjectSpamError(
+            _('Invalid project name. Spam detected. '
+                'Please email dev@readthedocs.org if this is an error.'))
```

Comment:
This is not likely to but much value to us. Especially if we are giving an error code of "spam detected" back to the user. Spammers will just shift patterns to something that works. We should instead be shadowbanning these users instead -- but I wouldn't recommend that with heuristic, it won't be accurate enough.